tests/mhd/test_integrated_mpi.py

```diff
@@ -167,7 +167,7 @@ def test_Residue_parallelization(self):
                 s = Spec(filename, mpi=mpi)
 
                 # Expand number of Fourier modes to include larger poloidal mode numbers:
-                s.boundary.change_resolution(6, s.boundary.ntor)
+                s.boundary = s.boundary.change_resolution(6, s.boundary.ntor)
                 # To make this example run relatively quickly, we will optimize in a
```

Comment:
What happens to the original boundary? Can you print the Optimizable graph and ensure that the old one is  garbage collected (or atleast not used).